cs/cs/Serializer/VowpalWabbitJsonSerializer.cs

```diff
@@ -0,0 +1,304 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="VowpalWabbitJsonSerializer.cs">
+//   Copyright (c) by respective owners including Yahoo!, Microsoft, and
+//   individual contributors. All rights reserved.  Released under a BSD
+//   license as described in the file LICENSE.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using VW.Interfaces;
+using VW.Labels;
+using VW.Serializer.Intermediate;
+
+namespace VW.Serializer
+{
+    /// <summary>
+    /// A deserializer from JSON to Vowpal Wabbit native examples.
+    /// </summary>
+    public sealed class VowpalWabbitJsonSerializer
+    {
+        internal const string FeatureIgnorePrefix = "_";
```

Comment:
why not put this in VowpalWabbitConstants and perhaps under a namespace/class called Serialization? Then you have all of these constant strings defined in the same place.